pkg/node/sendoperation/sendoperation.go

```diff
@@ -43,39 +46,28 @@ func message(expiry uint64, fee uint64, senderPubKey []byte, operation Operation
 	n = binary.PutUvarint(buf, expiry)
 	msg = append(msg, buf[:n]...)
 
-	// public key
-	msg = append(msg, senderPubKey...)
-
 	// operation
 	msg = append(msg, operation.Message()...)
 
 	return msg
 }
 
-func sign(msg []byte, privKey []byte) ([]byte, error) {
-	digest := blake3.Sum256(msg)
-
-	var auxBytes [32]byte
+func sign(msg []byte, pubKey []byte, privKey []byte) ([]byte, error) {
```

Comment:
It feel weird to pass a public function to a signature : can you append pubKey and msg before calling sign ?